pkg/controller/cronjob/cronjob_controllerv2.go

```diff
@@ -232,6 +232,7 @@ func (jm *ControllerV2) resolveControllerRef(namespace string, controllerRef *me
 	if controllerRef.Kind != controllerKind.Kind {
```

Comment:
IMO it would be good to add the group check here and test it. As mentioned here: https://github.com/kubernetes/kubernetes/pull/120184#discussion_r1315786267

UID check will stop this anyway, but it is just a small optimization

Reply:
If by group check you mean the kind of the `controllerRef`, then have added that change. Refer - https://github.com/kubernetes/kubernetes/pull/124222